wp-content/plugins/tw-acf-sync-helper/tw-acf-sync-helper.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Plugin Name: TW ACF Sync Helper
+ * Description: Syncs ACF fields semi automatically. Run "wp tw acf-sync-field-groups" to sync all ACF field groups.
+ */
+
+/**
+ * example usage wp-cli: wp tw acf-sync-field-groups
+ * example usage terminus: terminus wp the-world.{environemt} -- tw acf-sync-field-groups
+*/
+
+if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly
+
+/**
+ * Adds a wp cli command to sync all ACF field groups. Use it like this:
+ *
+ * "wp rhau acf-sync-field-groups"
+ *
+ */
+class ACFSyncFieldGroups {
+
+	/**
+	 * Init
+	 */
+	public static function init() {
+		add_action( 'acf/init', array( __CLASS__, 'add_wp_cli_command' ) );
+	}
+
+	/**
+	 * Conditional to check if inside WP_CLI
+	 *
+	 * @return boolean
+	 */
+	private static function is_wp_cli(): bool {
+		return defined( 'WP_CLI' ) && WP_CLI;
+	}
+
+	/**
+	 * Add the WP_CLI command
+	 *
+	 * @return void
+	 */
+	public static function add_wp_cli_command(): void {
+		if ( self::is_wp_cli() ) {
+			\WP_CLI::add_command( 'tw acf-sync-field-groups', array( __CLASS__, 'wp_cli_acf_sync_field_groups' ) );
+		}
+	}
+
+	/**
+	 * Syncs all ACF field groups
+	 *
+	 * ## OPTIONS
+	 *
+	 * @param array $args
+	 * @param array $assoc_args
+	 * @return void
+	 */
+	public static function wp_cli_acf_sync_field_groups( array $args = array(), array $assoc_args = array() ): void {
+		// Only allow this if invoked from WP CLI
+		if ( ! self::is_wp_cli() ) return;
+
+		acf_include( 'includes/admin/admin-internal-post-type-list.php' );
+
+		if ( ! class_exists( 'ACF_Admin_Internal_Post_Type_List' ) ) {
+			\WP_CLI::error( 'Some required ACF classes could not be found. Please update ACF to the latest version.' );
+		}
+
+		acf_include( 'includes/admin/post-types/admin-field-groups.php' );
+
+		/**
+		 * @var \ACF_Admin_Field_Groups $field_groups_class
+		 */
+		$field_groups_class = acf_get_instance( 'ACF_Admin_Field_Groups' );
+
+		$field_groups_class->setup_sync();
+
+		// Disable "Local JSON" controller to prevent the .json file from being modified during import.
+		acf_update_setting( 'json', false );
```

Comment:
Should we only need to do this if the settings is enable, and if so re-enable it after the import process is complete?

Reply:
Hi Rick, 

the function is just a shorthand to adding a filter in ACF. I've also already made some tests. Changing a field group will still update the .json file for it.

This is a reference to how the function is being used.
https://www.advancedcustomfields.com/resources/acf-settings

If the function would update anything permanently, it'll be strange for ACF Team to recommend running it in their plugin init.

Reply:
@rpeterman-gp The acf_update_setting is used to change the acf settings on load time, it is not firing any database update. So, the json setting will be disabled just until the wp-cli  process ends.